core/src/main/java/org/apache/iceberg/CatalogSupport.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.concurrent.ExecutorService;
+
+public interface CatalogSupport {
+
+  void withManifestAccessExecutor(ExecutorService executor);
+}
```

Comment:
This shouldn't be passed through the catalog or table operations. Instead, it should be passed through the scan interfaces by the processing engine. I think you probably want to add this to `TableScan` instead.

Reply:
Sounds good, i've updated the commits, PTAL.